web/src/settings_account.ts

```diff
@@ -575,9 +576,28 @@ export function set_up(): void {
         const data = {
             email: $("#change_email_form").find<HTMLInputElement>("input[name='email']").val(),
         };
+        const $status_element = $("#email-change-status").expectOne();
 
-        const opts = {
-            success_continuation() {
+        $status_element.fadeTo(0, 1);
+        loading.make_indicator($status_element, {text: settings_ui.strings.saving});
+
+        void channel.patch({
+            url: "/json/settings",
+            data,
+            success() {
+                setTimeout(() => {
+                    ui_report.message(
+                        $t_html(
+                            {
+                                defaultMessage:
+                                    "Check your email ({email}) to confirm the new address.",
+                            },
+                            {email: data.email},
+                        ),
+                        $status_element,
+                        "inline-block",
+                    );
+                }, 500);
```

Comment:
Why is this a `setTimeout`?

Reply:
Earlier, the email change process followed the use of `settings_ui.do_settings_change()`, which displayed the confirmation notice as a banner and included a checkmark after a 500ms delay using the `setTimeout` method.

However, the recent changes no longer use `settings_ui.do_settings_change()` due to the confirmation notice being displayed as a banner (with the checkmark) in the old implementation. The [issue](https://github.com/zulip/zulip/issues/31975) specifies updating the confirmation notice to a plain text line.

Therefore, I have attempted to replicate the behavior of `settings_ui.do_settings_change()` by displaying the confirmation notice after a 500ms delay using `setTimeout`.
Thank you.

Reply:
Great, this is a type of reasoned explanation that you should aim to provide for interesting decisions/details like this when opening PRs. It takes practice, but it's really helpful if one can explain the key decisions you made.